src/pathfinder/src/main.cc

```diff
@@ -26,41 +28,132 @@ struct CtlMsg
 class Pathfinder
 {
   std::vector<float> _angles;
+  std::vector<float> _filtered;
+#define lidar_steering
 
 public:
   Pathfinder() {}
 
   CtlMsg update_frame(sensor_msgs::LaserScan const& frame)
   {
-    if ((_angles.size() == 0) || (_angles[0] != frame.angle_min))
+    float _ranges = frame.ranges;
+    float _angle_min = frame.angle_min;
+    float _angle_max = frame.angle_max;
+    float _angle_increment = frame.angle_increment;
+    float _range_max = frame.range_max;
+    float _range_min = frame.range_min;
+
+    if ((_angles.size() == 0) || (_angles[0] != _angle_min[0]))
+    {
+      _angles.push_back(_angle_min[0]);
+      for (size_t i = 1; i < _ranges.size(); i++)
+        _angles.push_back(_angles.back() + _angle_increment[0]);
+    }
+
+    assert(_angles.size() == _ranges.size());
+
+#ifdef lidar_steering
+    for (size_t i = 0; i < _angles.size(); i++)
+    {
+      _filtered.push_back(_ranges[i]);
+    }
+
+    // Safety radius for car
+    float safety = 0.500 / 2.0;
+
+    for (int i = 0; i < (int)(_angles.size()); i++)
+    {
+      float _r = _ranges[i];
+
+      // Ensures distant points are filtered
+      if (_r > 10)
+      {
+        _r = 10.0;
+      }
+
+      float t = atan2(safety, _r);
```

Comment:
`t` is never modified, so it should be marked as `const`. Also `auto` should be used, in case we ever decide to use `double` or something.
```suggestion
      auto const t = atan2(safety, _r);
```

Reply:
Can you explain what auto does and when to use it a little more?

Reply:
auto should almost always be used when declaring a new local variable. It automatically detects the type of the value of the right-hand-side of the `=`. If you wand a detailed explanation of why it should almost always be used, Herb Sutter has a good one: https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/

Reply:
So is it proper to use const if a variable is being redefined every cycle, but isn't being changed after its definition?
For instance, here we are redefining t every cycle in the for loop, so I didn't think we were supposed to use const